v2/tools/asoctl/internal/importing/importable_arm_resource.go

```diff
@@ -167,30 +187,83 @@ func (i *importableARMResource) loader() extensions.ImporterFunc {
 		}
 
 		// Get the current status of the object from ARM
-		status, err := i.getStatus(ctx, i.armID, importable)
+		status, err := i.getStatus(ctx, i.armID.String(), importable)
 		if err != nil {
+			var responseError *azcore.ResponseError
```

Comment:
minor: You could probably stick this bit into some sort of `classifyError` function if you wanted (and/or if it needed to be reused elsewhere)? Might help the code read a bit cleaner too?